src/Table/Table.jsx

```diff
@@ -107,6 +113,12 @@ export class Table extends Component {
       newPages.push(pageData);
     }
 
+    if (this._epoch !== startEpoch) {
+      // the table was reset while we were fetching, so we don't want to
+      // actually insert our data
+      return;
+    }
```

Comment:
It doesn't really work here since there are multiple requests being made in a loop.